File: 3241-divide-array-into-arrays-with-max-difference/divide-array-into-arrays-with-max-difference.go
func divideArray(nums []int, k int) [][]int {
	sort.Ints(nums)
	n := len(nums)
	if n%3 != 0 {
		return [][]int{}
	}

	result := [][]int{}
	for i := 0; i < n; i += 3 {
		a, b, c := nums[i], nums[i+1], nums[i+2]
		if c-a > k {
			return [][]int{}
		}
		result = append(result, []int{a, b, c})
	}
	return result
}

File: 3241-divide-array-into-arrays-with-max-difference/divide-array-into-arrays-with-max-difference.rb
# @param {Integer[]} nums
# @param {Integer} k
# @return {Integer[][]}
def divide_array(nums, k)
  nums.sort!
  return [] if nums.size % 3 != 0

  result = []
  (0...nums.size).step(3) do |i|
    group = nums[i, 3]
    return [] if group[2] - group[0] > k
    result << group
  end
  result
end
